docs/src/proposals/interest-bearing-tokens.md

```diff
@@ -0,0 +1,383 @@
+vetted SPL token-conforming programs and group known mints by their program id.
+
+The token list will publish a file of known program ids at
+[token-list releases](https://github.com/solana-labs/token-list/releases/latest),
+to be used by RPC and programs.
+
+TODO consider adding Rust and C versions of the registry for on-chain programs and RPC
+
+TODO consider detailing the process of including a new program
+
+### Runtime
+
+#### Ephemeral Accounts
+
+In order to reduce friction when using read-only instructions on other programs,
+we need the ability for on-chain programs to dynamically create "ephemeral
+accounts", which only contain data, can be passed through CPI to other programs,
+and disappear after the instruction.
+
+For a program using i-tokens, to get the balance of a holding, the program creates
+an ephemeral account with 8 bytes of data, passes it to the i-token program's
+`GetHoldingBalance` instruction, then reads the information back as a `u64`. At the
+end of the instruction, the account disappears.
+
+TODO What's the best approach to implement this?  For example, does an ephemeral
+account even need a public key?
+
+This solution avoids the need to pass additional scratch accounts to almost all
+programs that use tokens.
+
+TODO what limits do we consider for these? A certain number of accounts?
+A total amount of bytes? Regarding security, this breaks the previous
+check that a CPI must use a subset of accounts provided to the calling program.
+
+#### Dynamic sysvars
+
+The i-token program also requires the use of dynamic sysvars. For example,
+during the `GetHoldingBalance` instruction, the program needs to know the current
+time in order to properly convert from token amount to share amount.
+
+Without dynamic sysvars, we need to create a system to tell what sysvars are
+needed for different instructions on each program, which creates complications
+for all users.
+
+### RPC
+
+The validator's RPC server needs to properly handle new token program accounts
+for read-only instructions and pre / post balances.
+
+#### `getBalance` and `getSupply`
+
+Instead of deserializing SPL token accounts and returning the data, the
+RPC server runs read-only transactions on the i-token program.
+
+RPC simply uses the same flow as on-chain programs: create an ephemeral account
+then runs the read-only transaction onto the bank requested.
+
+TODO how do we set the cost of read-only transactions? Do we lower it specially
+for read-only RPC calls? The i-token program will probably be one of the more
+cost-intensive computations possible since it is performing present value
+discounting.
+
+#### `preTokenBalances` and `postTokenBalances`
+
+As mentioned earlier in the Token Program Conformance section, the i-token
+program caches token balances (not share amounts) in the `amount` field of the
+SPL token holding. This should only be used by RPC, and not by programs to get
+the balance of a non-SPL holding.
```

Comment:
I don't think we should use this cached value it all. It might be out of date for preTokenBalances, and I don't think we can even guarantee its correctness for postTokenBalances unless we require every token account be loaded as writable (and rewritten) every transaction.

Reply:
The idea was to require programs to rewrite it on all transactions, since RPC will need access to it.  Do you think pre / post tokenBalances could call SerializeToSPL every time?  I'm not familiar with the flow there

Reply:
We discussed this last week, and we can have RPC use `SerializeToSPL` to fetch the correct balances, which eliminates the requirement to have the same layout